src/core/environment.js

```diff
@@ -1146,6 +1146,144 @@ function exitFullscreen() {
+ * of 3D objects.
+ *
+ * @method worldToScreen
+ * @param {p5.Vector} worldPosition The 3D coordinates in the world space.
+ * @return {p5.Vector} A vector containing the 2D screen coordinates.
+ * @example
+ * <div>
+ * <code>
+ * // Example 1: Convert 2D world coordinates of a rotating square to screen coordinates
+ * function setup() {
+ *   createCanvas(100, 100);
+ *
+ *   let vertices = [
+ *     createVector(-5, -5),
+ *     createVector(5, -5),
+ *     createVector(5, 5),
+ *     createVector(-5, 5)
+ *   ];
+ *
+ *   push(); // Start a new drawing state
+ *   translate(50, 50);
+ *   rotate(PI / 4);
+ *
+ *   // Convert each vertex to screen coordinates
+ *   let screenPos = vertices.map(v => worldToScreen(v));
+ *   pop(); // Restore original drawing state
+ *
+ *   background(200);
+ *
+ *   screenPos.forEach((pos, i) => {
+ *     text(`(${pos.x.toFixed(1)}, ${pos.y.toFixed(1)})`, pos.x, pos.y);
+ *   });
+ * }
+ * </code>
+ * </div>
+ * @example
+ * <div>
+ * <code>
+ * // Example 2: Convert 3D world coordinates of a rotating cube to 2D screen coordinates
+ * let vertices;
+ *
+ * function setup() {
+ *   createCanvas(100, 100, WEBGL);
+ *   vertices = [
+ *     createVector(-25, -25, -25),
+ *     createVector(25, -25, -25),
+ *     createVector(25, 25, -25),
+ *     createVector(-25, 25, -25),
+ *     createVector(-25, -25, 25),
+ *     createVector(25, -25, 25),
+ *     createVector(25, 25, 25),
+ *     createVector(-25, 25, 25)
+ *   ];
+ * }
+ *
+ * function draw() {
+ *   background(200);
+ *
+ *   // Animate rotation
+ *   let rotationX = millis() / 1000;
+ *   let rotationY = millis() / 1200;
+ *
+ *   push();
+ *
+ *   rotateX(rotationX);
+ *   rotateY(rotationY);
+ *
+ *   // Convert world coordinates to screen coordinates
+ *   let screenPos = vertices.map(v => worldToScreen(v));
+ *
+ *   pop();
+ *
+ *   // Display screen coordinates
+ *   screenPos.forEach((pos, i) => {
+ *
+ *     let screenX = pos.x - width / 2;
+ *     let screenY = pos.y - height / 2;
+ *     fill(255);
+ *     noStroke();
+ *     ellipse(screenX, screenY, 3, 3); // Draw points as small ellipses
+ *     fill(0);
+ *     text(`(${screenX.toFixed(1)}, ${screenY.toFixed(1)})`, screenX, screenY);
+ *   });
+ * }
+ * </code>
+ * </div>
+ *
+ */
+
+p5.prototype.worldToScreen = function(worldPosition) {
+  const renderer = this._renderer;
+  if (renderer.drawingContext instanceof CanvasRenderingContext2D) {
+    // Handle 2D context
+    const transformMatrix = new DOMMatrix()
+      .scale(1 / renderer._pInst.pixelDensity())
+      .multiply(renderer.drawingContext.getTransform());
+    const screenCoordinates = transformMatrix.transformPoint(
+      new DOMPoint(worldPosition.x, worldPosition.y)
+    );
+    return new p5.Vector(screenCoordinates.x, screenCoordinates.y);
+  } else {
+    // Handle WebGL context (3D)
+    const cameraCoordinates = renderer.uMVMatrix.multiplyPoint(worldPosition);
```

Comment:
oh, maybe it's this -- we split up `uMVMatrix` into `uModelMatrix` and `uViewMatrix`, and only update `uMVMatrix` right before we draw a shape here: https://github.com/processing/p5.js/blob/4fbbaf514856ec11b771f701dab5dea7e7f14267/src/webgl/p5.Shader.js#L579-L584

Maybe we can refactor the bit that sets `uMVMatrix` into a function on the renderer, like `calculateCombinedMatrix()`, which gets called in `_setMatrixUniforms` above, and then you can also call it here right before you make `cameraCoordinates`?